src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -210,16 +211,16 @@ AssemblyMetadata GetMetadataWorker()
                     StreamCopy(fileStream, copyStream, offset, size);
                 }
 
-                // copy over the data to temp storage and let pooled stream go
-                storage = _temporaryStorageService.CreateTemporaryStreamStorage();
-
+                // Now, copy over the metadata bytes into a memory mapped file.  This will keep it fixed in a single
+                // location, so we can create a metadata value wrapping that.  This will also let us share the memory
+                // for that metadata value with our OOP process.
                 copyStream.Position = 0;
-                storage.WriteStream(copyStream, CancellationToken.None);
+                storageHandle = _temporaryStorageService.WriteToTemporaryStorage(copyStream, CancellationToken.None);
             }
 
-            // get stream that owns the underlying unmanaged memory.
-            stream = storage.ReadStream(CancellationToken.None);
-
+            // Now, read the data from the memory-mapped-file back into a stream that we load into the metadata value.
+            stream = _temporaryStorageService.ReadFromTemporaryStorageService(storageHandle.Identifier, CancellationToken.None);
+            GC.KeepAlive(storageHandle);
```

Comment:
>GC.KeepAlive(storageHandle); [](http://example.com/codeflow?start=12&length=28)

Can you explain the need for this a bit? It's referenced on the previous line so it doesn't fit the model of when I would think this would be needed.

Reply:
removed.  not needed here.